examples/step-37/step-37.cc

```diff
@@ -168,228 +170,137 @@ namespace Step37
-  // operator application that is integrated in deal.II. The main class that
-  // collects all data is called MatrixFree. It contains mapping information
-  // (Jacobians) and index relations between local and global degrees of
-  // freedom. It also contains constraints like the ones from Dirichlet
-  // boundary conditions (or hanging nodes, if we had any). Moreover, it can
-  // issue a loop over all cells in %parallel, where it makes sure that only
-  // cells are worked on that do not share any degree of freedom (this makes
-  // the loop thread-safe when writing into destination vectors). This is a
-  // more advanced strategy compared to the WorkStream class described in the
-  // @ref threads module that serializes operations that might not be
-  // thread-safe. Of course, to not destroy thread-safety, we have to be
-  // careful when writing into class-global structures.
+  // you can apply it to a vector (the <code>vmult()</code> function). The
+  // difference to a real matrix of course lies in the fact that this class
+  // does not actually store the <i>elements</i> of the matrix, but only knows
+  // how to compute the action of the operator when applied to a vector.
   //
-  // First comes the implementation of the matrix-free class. It provides some
-  // standard information we expect for matrices (like returning the
-  // dimensions of the matrix), it implements matrix-vector multiplications in
-  // several forms (transposed and untransposed), and it provides functions
-  // for initializing the structure with data. The class has three template
-  // arguments, one for the dimension (as many deal.II classes carry), one for
-  // the degree of the finite element (which we need to enable efficient
-  // computations through the FEEvaluation class), and one for the underlying
-  // scalar type. We want to use <code>double</code> numbers (i.e., double
-  // precision, 64-bit floating point) for the final matrix, but floats
-  // (single precision, 32-bit floating point numbers) for the multigrid level
-  // matrices (as that is only a preconditioner, and floats can be worked with
-  // twice as fast).
+  // The infrastructure describing the matrix size, the initialization from a
+  // MatrixFree object, and the various interfaces to matrix-vector products
+  // through vmult() and Tvmult() methods, is provided by the class
+  // MatrixFreeOperator::Base from which this class derives. The
+  // LaplaceOperator class defined here only has to provide a few interfaces,
+  // namely the actual action of the operator through the apply_add() method
+  // that gets used in the vmult() functions, and a method to compute the
+  // diagonal entries of the underlying matrix. We need the diagonal for the
+  // definition of the multigrid smoother. Since we consider a problem with
+  // variable coefficient, we further implement a method that can fill the
+  // coefficiient values.
```

Comment:
maybe mention that there is already `MatrixFreeOperators::LaplaceOperator`, but you re-do it here for educational purposes.